pkg/utils/test/matchers/jq/jq_transform.go

```diff
@@ -32,3 +32,36 @@ func Extract(expression string) func(in any) (any, error) {
 		return v, nil
 	}
 }
+
+func ExtractValue[T any](in any, expression string) (T, error) {
+	var result T
+	var ok bool
+
+	query, err := gojq.Parse(expression)
+	if err != nil {
+		return result, fmt.Errorf("unable to parse expression %s, %w", expression, err)
+	}
+
+	data, err := toType(in)
+	if err != nil {
+		return result, err
+	}
+
+	it := query.Run(data)
+
+	v, ok := it.Next()
+	if !ok {
+		return result, nil
+	}
+
+	if err, ok := v.(error); ok {
+		return result, err
+	}
+
```

Comment:
is it basically ... v, err := Extract(expression); if err .. ; result, ok := v.(T) ..  ?

Are these functions documented somewhere?

Reply:
yep, probably need the other way: using `ExtractValue` in `Extract`, wdyt ?

Reply:
done

Reply:
> yep, I can probably refactor the other way: using `ExtractValue` in `Extract`, wdyt ?

Still in progress, will share thoughts if any after get more context